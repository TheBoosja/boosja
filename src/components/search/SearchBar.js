import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import { Form } from 'components/styled';

export class SearchBar extends Component {
	constructor(props) {
		super(props);

		this.onFormSubmit = this.onFormSubmit.bind(this);
	}

	onFormSubmit({ query }) {
		const { history, match } = this.props;

		history.push(`${match.url}/${query}`);
	}

	render() {
		const { handleSubmit } = this.props;

		return (
			<Form onSubmit={handleSubmit(this.onFormSubmit)}>
				<Field
					name="query"
					placeholder="Search"
					autoFocus
					component="input"
				/>
			</Form>
		);
	}
}

function validate({ query }) {
	return !query ? { query: 'Enter a query' } : {};
}

const ComputedSearchBar = reduxForm({ form: 'search', validate })(SearchBar);

ComputedSearchBar.displayName = 'SearchBar';
export default ComputedSearchBar;
