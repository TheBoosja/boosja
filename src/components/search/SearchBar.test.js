import React from 'react';
import { shallow } from 'enzyme';
import { SearchBar } from './SearchBar';

describe('Search Bar', () => {
	const mockPush = jest.fn();
	const mockHandleSubmit = jest.fn();

	const props = { 
		history: { push: mockPush },
		match: { url: '/search' },
		handleSubmit: mockHandleSubmit
	};
	const searchBar = shallow(<SearchBar {...props} />);

	it('renders properly', () => {
		expect(searchBar).toMatchSnapshot();
	});

	it('contains a Form component', () => {
		expect(searchBar.find('Form').exists()).toBe(true);
	});

	it('contains a Field component', () => {
		expect(searchBar.find('Field').exists()).toBe(true);
	});

	describe('when user submits search query', () => {
		it('redirects to query route', () => {
			const query = { query: 'lady' };
			searchBar.instance().onFormSubmit(query);

			expect(mockPush).toHaveBeenCalledWith(`/search/${query.query}`);
		});
	});
});