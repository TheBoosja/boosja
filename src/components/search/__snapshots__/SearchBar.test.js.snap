// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    handleSubmit={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    match={
        Object {
            "url": "/search",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Field
        autoFocus={true}
        component={[Function]}
        name="query"
        placeholder="Search"
/>,
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "component": [Function],
        "name": "query",
        "placeholder": "Search",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Field
          autoFocus={true}
          component={[Function]}
          name="query"
          placeholder="Search"
/>,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "component": [Function],
          "name": "query",
          "placeholder": "Search",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
